src/GuzzleRequestAdapter.php

```diff
@@ -81,16 +82,36 @@ public function __construct(AuthenticationProvider $authenticationProvider,
         $this->guzzleClient = ($guzzleClient) ?: KiotaClientFactory::create();
     }
 
+    /**
+     * @param RequestInformation $requestInfo
+     * @param ResponseInterface $result
+     * @param array<string,array{string,string}>|null $errorMappings
+     * @return Promise|null
+     */
+    private function tryHandleResponse(RequestInformation $requestInfo, ResponseInterface $result, ?array $errorMappings = null): ?Promise
+    {
+        $responseHandlerOption = $requestInfo->getRequestOptions()[ResponseHandlerOption::class] ?? null;
+        if ($responseHandlerOption && is_a($responseHandlerOption, ResponseHandlerOption::class)) {
+            $responseHandler = $responseHandlerOption->getResponseHandler();
+            if ($responseHandler !== null) {
```

Comment:
Thinking the `ResponseHandler` in the [ResponseHandlerOption](https://github.com/microsoft/kiota-http-guzzle-php/blob/dev/src/Middleware/Options/ResponseHandlerOption.php#L34) shouldn't be nullable in the first place.